pkg/adaptor/cloud/aws/provider.go

```diff
@@ -186,3 +203,63 @@ func (p *awsProvider) DeleteInstance(ctx context.Context, instanceID string) err
 func (p *awsProvider) Teardown() error {
 	return nil
 }
+
+// Add SelectInstanceType method to select an instance type based on the memory and vcpu requirements
+func (p *awsProvider) SelectInstanceType(ctx context.Context, memory int64, vcpu int64) (string, error) {
+
+	return cloud.GetBestFitInstanceType(p.serviceConfig.InstanceTypesTupleList, memory, vcpu)
+}
+
+// Add a method to populate InstanceTypesTupleList for all the instanceTypes
+func (p *awsProvider) updateInstanceTypesTupleList() error {
+
+	// Get the instance types from the service config
+	instanceTypes := p.serviceConfig.InstanceTypes
+
+	// If instanceTypes is empty then populate it with the default instance type
+	if len(instanceTypes) == 0 {
+		instanceTypes = append(instanceTypes, p.serviceConfig.InstanceType)
+	}
+
+	// Create a list of instance type tuples
+	var instanceTypesTupleList []cloud.InstanceTypeTuple
+
+	// Iterate over the instance types and populate the instanceTypesTupleList
+	for _, instanceType := range instanceTypes {
+		vcpu, memory, err := p.getInstanceTypeInformation(instanceType)
```

Comment:
This could error because an specified instance-type isn't available in the region, in that situation would it be preferable to continue/skip that instance type rather return and not return a valid provider at all?

Reply:
Or we assume that admin is correct and returning the error will force the admin to provide the correct instance types ?

Reply:
@jtumber-ibm I am tending to favour keeping the code as-is, ie returning an error if there is a failure in populating the details of the instance type . The error gives an indication to the admin of what is wrong with the input instance types (like unavailability, typo etc)
And since the error will be during the start of CAA, it'll be a good indication of a wrong config as well.